src/plugins/orangehrmAuthenticationPlugin/Auth/LocalAuthProvider.php

```diff
@@ -36,14 +47,36 @@ private function getAuthenticationService(): AuthenticationService
     }
 
     /**
-     * @inheritDoc
+     * @param AuthParamsInterface $authParams
+     * @return bool
+     * @throws AuthenticationException
+     * @throws PasswordEnforceException
      */
     public function authenticate(AuthParamsInterface $authParams): bool
     {
         if (!$authParams->getCredential() instanceof UserCredentialInterface) {
             return false;
         }
-        return $this->getAuthenticationService()->setCredentials($authParams->getCredential());
+        $success = $this->getAuthenticationService()->setCredentials($authParams->getCredential());
+        if ($success) {
+            if ($this->getConfigService()->getConfigDao()
+                    ->getValue(ConfigService::KEY_ENFORCE_PASSWORD_STRENGTH) === 'on') {
+                $passwordStrengthValidation = new PasswordStrengthValidation();
+                $passwordStrength = $passwordStrengthValidation->checkPasswordStrength(
+                    $authParams->getCredential()->getPassword()
+                );
+
+                if (!($this->getPasswordStrengthService()
+                    ->isValidPassword($authParams->getCredential()->getPassword(), $passwordStrength))
```

Comment:
```suggestion
                    ->isValidPassword($authParams->getCredential(), $passwordStrength))
```
avoid passing plain password to functions